test/functional/test_framework/miniscript.py

```diff
@@ -0,0 +1,1709 @@
+        return [child_dsat_set[1]+child_dsat_set[0]]
+
+    def _or_c_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0],
+                child_sat_set[1]+child_dsat_set[0]]
+
+    def _or_c_dsat(self, child_sat_set, child_dsat_set):
+        return [[]]
+
+    def _or_i_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]+[(SatType.DATA, b'\x01')],
+                child_sat_set[1]+[(SatType.DATA, b'')]]
+
+    def _or_i_dsat(self, child_sat_set, child_dsat_set):
+        return [[(SatType.DATA, b'\x01')]+child_dsat_set[1] + [
+                (SatType.DATA, b'')]]
+
+    def _andor_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[1]+child_sat_set[0],
+                child_sat_set[2]+child_dsat_set[0]]
+
+    def _andor_dsat(self, child_sat_set, child_dsat_set):
+        return [child_dsat_set[2]+child_dsat_set[0]]
+
+    def _a_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _a_dsat(self, child_sat_set, child_dsat_set):
+        return [child_dsat_set[0]]
+
+    def _s_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _s_dsat(self, child_sat_set, child_dsat_set):
+        return [child_dsat_set[0]]
+
+    def _c_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _c_dsat(self, child_sat_set, child_dsat_set):
+        return [child_dsat_set[0]]
+
+    def _t_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _t_dsat(self, child_sat_set, child_dsat_set):
+        return [[]]
+
+    def _d_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]+[(SatType.DATA, b'\x01')]]
+
+    def _d_dsat(self, child_sat_set, child_dsat_set):
+        return [[(SatType.DATA, b'')]]
+
+    def _v_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _v_dsat(self, child_sat_set, child_dsat_set):
+        return [[]]
+
+    def _j_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _j_dsat(self, child_sat_set, child_dsat_set):
+        return [[(SatType.DATA, b'')]]
+
+    def _n_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]]
+
+    def _n_dsat(self, child_sat_set, child_dsat_set):
+        return [child_dsat_set[0]]
+
+    def _l_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]+[(SatType.DATA, b'')]]
+
+    def _l_dsat(self, child_sat_set, child_dsat_set):
+        return [[(SatType.DATA, b'\x01')]]
+
+    def _u_sat(self, child_sat_set, child_dsat_set):
+        return [child_sat_set[0]+[(SatType.DATA, b'\x01')]]
+
+    def _u_dsat(self, child_sat_set, child_dsat_set):
+        return [[(SatType.DATA, b'')]]
+
+    def _thresh_sat(self, child_sat_set, child_dsat_set):
+        thresh_sat_ls = []
+        n = len(self.children)
+        for i in range(2**n):
+            if bin(i).co
```

Comment:
why not `child.t.name.startswith('WRAP_')` ?

Reply:
That's much better. Thx.